numpy/core/src/common/npy_config.h

```diff
@@ -81,6 +81,55 @@
 #endif
 #endif /* defined(_MSC_VER) && defined(__INTEL_COMPILER) */
 
+/* Disable functions that give undesired floating-point outputs on
+   Cygwin. */
+#ifdef __CYGWIN__
+
+#undef HAVE_CABS
+#undef HAVE_CABSF
+#undef HAVE_CABSL
+
+/* These work for me in C++, even for the failing test case (2 ** 29),
+   but the tests fail.  No idea if the functions I'm getting are used
+   or not. */
+#undef HAVE_LOG2
+#undef HAVE_LOG2F
+#undef HAVE_LOG2L
+
+/* Complex square root does not use sign of zero to find branch
+   cuts. */
+#undef HAVE_CSQRT
+#undef HAVE_CSQRTF
+#undef HAVE_CSQRTL
+
+/* C++ Real asinh works fine, complex asinh says asinh(1e-20+0j) =
+   0 */
+#undef HAVE_CASINH
+#undef HAVE_CASINHF
+#undef HAVE_CASINHL
+
+/* _check_branch_cut(np.arcsin, ...) fails */
+#undef HAVE_CASIN
+#undef HAVE_CASINF
+#undef HAVE_CASINL
+
+/* Branch cuts for arccosine also fail */
+#undef HAVE_CACOS
+#undef HAVE_CACOSF
+#undef HAVE_CACOSL
+
+/* Branch cuts for arctan fail as well */
+#undef HAVE_CATAN
+#undef HAVE_CATANF
+#undef HAVE_CATANL
+
+/* check_loss_of_precision fails in arctanh */
+#undef HAVE_CATANH
+#undef HAVE_CATANHF
+#undef HAVE_CATANHL
+
+#endif
+
```

Comment:
Are these due to cygwin using an old glibc? If so, maybe we can tie it to the glibc version like below.

Reply:
Cygwin is based on Newlib, not GNU C-library.  If glibc is something different I can look into checking that.

The list of problem functions here looks very similar to the one for MSVC and MinGW; I think acos and friends are the only ones in the Windows list but not the Cygwin one.  I may consolidate those lists.

Reply:
The `__CYGWIN__` macro unfortunately does not include a version.

The `modfl` code in cygwin is taken from the MinGW-w64 project, so the inverse trigonometric functions working in Cygwin may mean that they're working on Windows and can be removed from that list.  I don't have a Windows build environment, so it will be a while before I could try that.